Make Base.shell_escape() and Base.shell_split() public

JamesWrigley opened the pull request:
`Base.shell_escape()` is particularly useful since it's the only way (AFAICT) to properly convert a `Cmd` to a string.

JamesWrigley committed:
Make Base.shell_escape() and Base.shell_split() public

Base.shell_escape() is particularly useful since it's the only way to properly
convert a Cmd to a string.

JamesWrigley commented:
(bump)

StefanKarpinski commented:
I'm definitely on board with making them public. Should we also export them?

JamesWrigley commented:
Hmm to me they're not quite reached-for-often-enough to be worth exporting, but I don't feel very strongly about it.

StefanKarpinski commented:
Yeah, that's fair. Will discuss next triage call, but that shouldn't block merging this as is.

oscardssmith commented:
triage says public is good.

Seelengrab commented:
Can this be done for the other platform specific shell escapes as well?

JamesWrigley commented:
Do you mean these?
- `Base.shell_escape_posixly()`
- `Base.shell_escape_csh()`
- `Base.shell_escape_wincmd()`
- `Base.escape_microsoft_c_args()` (slightly different in that it takes a list of arguments and returns a single string)

Seelengrab commented:
Yes, exactly. Their docstrings are very in-depth, and I was planning on putting them in the manual in some form so that the docstrings of ` Cmd`/`run` can reference them, to provide an anchor point for people that e.g. have to call `cmd.exe` directly (which uses special parsing of arguments, so should be aware of the tradeoffs).

JamesWrigley committed:
Make a bunch of other command/string escape functions public

Note that Base.escape_raw_string() is already in the manual so it should be
public anyway.

JamesWrigley commented:
Sure, I guess that makes sense. Made them public in 5d89485bb9, along with `Base.escape_raw_string()` because it's referenced by one of the other docstrings (and is actually already in the manual so it should be public anyway).

JamesWrigley commented:
(bump)
